### PR TITLE
fix: tear down Node's per-process state before the process exits

### LINKED ISSUE #52860: Flaky test: nn-test — short tls/https Node tests die with a silent SIGABRT (exitcode -6)
Random tests in the Node.js suite (`linux-x64 / nn-test / Run Node.js Tests`) die with a silent SIGABRT — no stderr, no stack — and pass 9/9 on the `--measure-flakiness` rerun, which still fails the job. The victim is a different test each time but always one that loads `tls`/`https` and exits quickly.

**Test:** any short `parallel/test-tls-*` / `parallel/test-https-*` (seen: `test-tls-get-ca-certificates-extra`, `test-tls-get-ca-certificates-extra-empty`, `test-https-argument-of-creating`)
**Job:** `linux-x64 / nn-test / Run Node.js Tests` (`script/node-spec-runner.js`)
**Platform:** linux x64 (the bug is cross-platform, this is just where we run the suite)
**Frequency:** 3 of the last 34 completed nn-test jobs (2026-08-14 → 08-16), all on unrelated PR branches, each `failed 1 out of 10` on rerun. Locally ~1 in 30k test processes at `-j128`; `ELECTRON_RUN_AS_NODE=1 electron -e "require('tls')"` hits it ~1 in 700.
**Error / signal:** `severity: crashed` / `exitcode: -6`, or from `test/common/child_process.js`: `status = null, signal = SIGABRT` with empty `--- stderr ---` / `--- stdout ---`.

**Failing runs:**
- https://github.com/electron/electron/actions/runs/31939048118/job/95146708674 (`test-tls-get-ca-certificates-extra`)
- https://github.com/electron/electron/actions/runs/31934729186/job/95136834082 (`test-https-argument-of-creating`)
- https://github.com/electron/electron/actions/runs/31909127762/job/95072968384 (`test-tls-get-ca-certificates-extra-empty`)

**Cause:** since Node 24.9 (nodejs/node#59856) `require('tls')` starts a `uv_thread` that loads the CA certificates and then takes `per_process::cli_options_mutex`. Node joins it in `node::TearDownOncePerProcess()`, which we never call, so if the process exits first the static mutex is destroyed by `exit()` under the thread and libuv's `uv_mutex_lock` `abort()`s. Core from a local repro:

```
main thread:                          crashing thread:
  exit() → _dl_fini (static dtors)      abort()
  __libc_start_main                     uv_mutex_lock()               deps/uv/src/unix/thread.c:366
                                        node::crypto::LoadCACertificates()  src/crypto/crypto_context.cc:934
```

Not limited to `ELECTRON_RUN_AS_NODE`: `require('node:tls'); app.exit()` before `ready` in the browser process aborts ~1 in 5. All of 41–44-x-y ship Node 24.18 and are affected.

### PR DESCRIPTION
**Pull request by MarshallOfSound**
#### Description of Change

Since Node 24.9 (nodejs/node#59856), loading `tls` starts a background thread that parses the CA certificates. Upstream joins it in `node::TearDownOncePerProcess()`, but Electron never called that on any exit path, so a process that loads `tls`/`https` and exits soon after can have `exit()` destroy Node's static `per_process::cli_options_mutex` underneath that thread → `uv_mutex_lock` → `abort()`. Silent SIGABRT, no output. This is what has been picking off random `test-tls-*` / `test-https-*` tests in the `nn-test` job (3 of the last 34 runs). Fixes #52860.

- `NodeMain` (`ELECTRON_RUN_AS_NODE`): call `node::TearDownOncePerProcess()` after `V8::Dispose()`, matching `node::Start()`.
- `NodeBindings`: the instance that ran `InitializeOncePerProcess()` now runs `TearDownOncePerProcess()` in its destructor — covers normal browser quit and the utility process.
- `Browser::ExitWithCode`: the pre-ready `app.exit()` path calls `exit()` directly with nothing torn down, so tear Node down first there too.
- spec: `require('node:tls'); app.exit(123)` before ready, 15×, asserting a clean exit.

Linux x64 testing build, this branch vs `main`:

| scenario | before | after |
|---|---|---|
| `require('node:tls'); app.exit()` before ready (browser process) | 20/100 SIGABRT, 37/100 read destroyed statics | 0/700 |
| `ELECTRON_RUN_AS_NODE=1 electron -e "require('tls')"` | 3/2,000 SIGABRT | 0/20,000 |
| Node `parallel/test-tls-*` at `-j128` | 3/96,000 (`exitcode: -6`, all three cores in `LoadCACertificates → uv_mutex_lock → abort`) | 0/48,000 |

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [x] relevant API documentation, tutorials, and examples are updated and follow the [documentation style guide](https://github.com/electron/electron/blob/main/docs/development/style-guide.md)
- [x] [PR release notes](https://github.com/electron/clerk/blob/main/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/main/README.md#examples).

#### Release Notes

Notes: Fixed a possible crash (SIGABRT) during process exit when the process had loaded `tls`/`https` shortly before exiting, affecting `app.exit()` before `ready` and short-lived `ELECTRON_RUN_AS_NODE` / `child_process.fork()` scripts.

### CODE REVIEW

**Review by codebytere — APPROVED** [review]

File: shell/app/node_main.cc [hunk 1]
```diff
@@ -329,6 +329,11 @@ int NodeMain() {
 
   v8::V8::Dispose();
 
+  // Matches node::InitializeOncePerProcess() above. In particular this joins
+  // the off-thread CA certificate loader that `require('tls')` starts, which
+  // would otherwise race the static destructors run by exit() and abort().
+  node::TearDownOncePerProcess();
```
Comment (deepak1556) [inline shell/app/node_main.cc:335]: nit: `node::ResetStdio` call a few lines above can now be dropped in favor of this.
